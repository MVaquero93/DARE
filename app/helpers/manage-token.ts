import axios from 'axios'
import envConfig from '../config/env'
import { TokenData } from '../interfaces/tokenData.interface'
import { login } from '../services/login.service'
let data = {} as TokenData

export const getToken = async () => {
  return data
}

export const setToken = async () => {
  try {
    data = await login()

    setTimeout(() => {
      data = {} 
    }, envConfig.credentials.expires_in)

    return data
  } catch (err) {
    console.error(err)
    return null
  }
}