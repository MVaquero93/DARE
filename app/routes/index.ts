import express from 'express'
import apiRouter from './api'


const router = express.Router()

/**
 * load api routes
 */
router.use('/api', apiRouter)


export default router
