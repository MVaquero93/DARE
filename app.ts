import express from 'express'
import envConfig from './app/config/env'
import routes from './app/routes'
import bodyParser from 'body-parser'

const app = express()


app.use(
  bodyParser.urlencoded({extended: false}),
  bodyParser.json(),
)
app.use(routes)

app.get('/', (req, res) => {
  res.send('Welcome to DARE')
})

const server = app.listen(envConfig.app.port, () => {
  console.log('listening at port: ' + envConfig.app.port)
})

export default server
