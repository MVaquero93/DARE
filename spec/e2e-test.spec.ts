import { setToken } from '../app/services/manage-token.service'
import app from '../app'
import request from 'request-promise-native';
import envConfig from '../app/config/env';

describe('e2e testing example', () => {

  const endPoint = envConfig.app.protocol + '://' + envConfig.app.domain + ':' + envConfig.app.port

  beforeAll(async (done) => {
    await setToken()
    done()
  })

  afterAll((done) => {
    app.close(() => {
      done()
    })
  })

  it('Call to clients list endpoint', async () => {
    const resp = await request({
      uri: endPoint + '/api/v1/clients',
      method: 'GET',
      resolveWithFullResponse: true,
      simple: false,
      json: true,       
    })
    expect(resp.data).toMatchSnapshot()
  })
})