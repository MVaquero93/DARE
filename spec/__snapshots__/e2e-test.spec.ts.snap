// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e testing example Call to clients list endpoint 1`] = `undefined`;
